common/SmokeTests/SmokeTest/Update-Dependencies.ps1

```diff
@@ -31,63 +24,74 @@ function Log-Warning($message) {
     }
 }
 
-Register-PackageSource `
-    -Name $NIGHTLY_FEED_NAME `
-    -Location $NIGHTLY_FEED_URL `
-    -ProviderName Nuget `
-    -ErrorAction SilentlyContinue `
+function GetAllPackages {
+    $packages = Find-Package -Source $PACKAGE_FEED_URL -AllVersion -AllowPrereleaseVersions
+    if ($Daily) {
+        return $packages | Where-Object { $_.Version.Contains("alpha") }
+    }
+    return $packages | Where-Object { !$_.Version.Contains("alpha") -and !$_.Version.Contains("dev") }
+}
 
-# List all packages from the source specified by $FeedName. Packages are sorted
-# ascending by version according to semver rules (e.g. 4.0.0-preview.1 comes
-# before 4.0.0) not lexicographically.
-# Packages cannot be filtered at this stage because the sleet feed to which they
-# are published does not support filtering by name.
-$allPackages = Find-Package -Source $NIGHTLY_FEED_NAME -AllVersion -AllowPrereleaseVersions
+function GetLatestPackage([array]$packageList, [string]$packageName) {
+    $versions = ($packageList
+        | Where-Object { $_.Name -eq $packageName }
+        | Select-Object -ExpandProperty Version)
 
-$baselineVersionDate = $null;
+    if (!$versions) {
+        Write-Warning "Did not find any versions for $($packageName)"
+        return
+    }
 
-# For each PackageReferecne in the csproj, find the latest version of that
-# package from the dev feed which is not in the excluded list.
-$projectFilePath = Resolve-Path -Path $ProjectFile
-[xml]$csproj = Get-Content $ProjectFile
-$csproj |
-    Select-XML $PACKAGE_REFERENCE_XPATH |
-    Where-Object { $_.Node.HasAttribute('Version') } |
-    ForEach-Object {
-        # Resolve package version:
-        $packageName = $_.Node.Include
-
-        # This assumes that the versions coming back from Find-Package are
-        # sorted ascending. It excludes any version that is in the corresponding
-        # $PACKAGE_EXCLUSIONS entry
-        $targetVersion = ($allPackages |
-            Where-Object { $_.Name -eq $packageName } |
-            Where-Object { -not ( $PACKAGE_EXCLUSIONS.ContainsKey($packageName) -and $PACKAGE_EXCLUSIONS[$packageName].ContainsKey($_.Version)) } |
-            Select-Object -Last 1).Version
-
-        if ($targetVersion -eq $null) {
-            return
-        }
+    $sorted = [AzureEngSemanticVersion]::SortVersionStrings($versions)
+    return $sorted | Select-Object -First 1
+}
 
-        Write-Host "Setting $packageName to $targetVersion"
-        $_.Node.Version = "$targetVersion"
+function SetLatestPackageVersions([xml]$csproj) {
+    # For each PackageReference in the csproj, find the latest version of that
+    # package from the dev feed which is not in the excluded list.
+    $allPackages = GetAllPackages
+    $csproj |
+        Select-XML $PACKAGE_REFERENCE_XPATH |
+        Where-Object { $_.Node.HasAttribute('Version') } |
+        Where-Object { -not $_.Node.HasAttribute('DoNotUpdate') }
```

Comment:
In the case of a !Daily build we should make sure we still update this in the case where we are about to release it, and we find it in the artifacts. I know you are updating those elsewhere currently but I just gave a comment about updating those in this script as well so I wanted to make sure you thought about this case.

Reply:
I think I can just skip the override when `!Daily`, because we only get the "version referenced is older" error from dotnet restore when the alpha version is on the same major/minor as the beta/ga version, for the one specific override this is for (eventhubs 5.3.0-beta.4).